### PR TITLE
Pagination feature

### PR DESCRIPTION
**Pull request by Vahram1995**

### CODE REVIEW

File: src/Libraries/Database/Sleekdb/Paginator.php [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+	/**
+	 * @param bool $withBaseUrl
+	 * @return string|null
+	 */
+	public function previousPageLink(bool $withBaseUrl = false): ?string
+	{
+		$previous = null;
+		if (!empty($this->previousPageNumber())) {
+			$previous = $this->getUri($withBaseUrl) . 'per_page=' . $this->per_page . '&page=' . $this->previousPageNumber();
```
Comment (andrey-smaelov) [inline src/Libraries/Database/Sleekdb/Paginator.php:104]: `per_page` and `page` are spread over the code where can be used as constants

File: src/Libraries/Database/Sleekdb/Paginator.php [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+	/**
+	 * @var SleekDbal
+	 */
+	private $dbal;
+
+	/**
+	 * @var int
+	 */
+	protected $per_page;
```
Comment (andrey-smaelov) [inline src/Libraries/Database/Sleekdb/Paginator.php:27]: Variables and properties are always should be in camel case

File: src/Libraries/Database/Idiorm/Paginator.php [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+		}
+		return $last;
+	}
+
+	/**
+	 * @return mixed
+	 */
+	public function firstItem()
+	{
+		return $this->data[array_key_first($this->data)];
```
Comment (armanist) [inline src/Libraries/Database/Idiorm/Paginator.php:164]: It seems like` $this->data` can also be of type `IdiormResultSe`t; however, parameter `$array` of `array_key_first()` does only seem to accept array, maybe add an additional type check?

File: src/Libraries/Database/Idiorm/Paginator.php [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+	}
+
+	/**
+	 * @return mixed
+	 */
+	public function lastItem()
+	{
+		return $this->data[array_key_last($this->data)];
```
Comment (armanist) [inline src/Libraries/Database/Idiorm/Paginator.php:172]: It seems like `$this->data` can also be of type `IdiormResultSet`; however, parameter `$array` of `array_key_last()` does only seem to accept array, maybe add an additional type check?

File: src/Libraries/Database/Sleekdb/Paginator.php [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+
+		return $links;
+	}
+
+	/**
+	 * @param bool $withBaseUrl
+	 * @param $pageItemsCount
+	 * @return string|null
+	 */
+	public function getPagination(bool $withBaseUrl = false, $pageItemsCount = null): ?string
```
Comment (armanist) [inline src/Libraries/Database/Sleekdb/Paginator.php:217]: `getPagination()` method is defined in 2 places [code duplicate]

Comment (armanist) [reply src/Libraries/Database/Sleekdb/Paginator.php:217]: `getPagination()`  too much conditions, complexity is very high

**Review by armanist — CHANGES_REQUESTED** [review]

**Review by armanist — APPROVED** [review]

**Review by Arno2005 — APPROVED** [review]

**Pull request merged**